File: docs-examples/example-java/src/test/java/io/micronaut/ignite/docs/config/IgniteCacheConfigurationFactory.java
package io.micronaut.ignite.docs.config;

import io.micronaut.context.annotation.Bean;
import io.micronaut.context.annotation.Factory;
import io.micronaut.ignite.annotation.IgnitePrimary;
import org.apache.ignite.configuration.CacheConfiguration;

@Factory
public class IgniteCacheConfigurationFactory {
    @Bean
    @IgnitePrimary
    CacheConfiguration<String,String> cacheInstance() {
        return new CacheConfiguration<String, String>("my-cache")
            .setBackups(10);
    }
}

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/ignite/docs/config/IgniteCacheConfigurationFactory.kt
package io.micronaut.ignite.docs.config

import io.micronaut.context.annotation.Bean
import io.micronaut.context.annotation.Factory
import io.micronaut.ignite.annotation.IgnitePrimary
import org.apache.ignite.configuration.CacheConfiguration

@Factory
class IgniteCacheConfigurationFactory {
    @Bean
    @IgnitePrimary
    fun cacheInstance(): CacheConfiguration<String, String>? {
        return CacheConfiguration<String, String>("my-cache")
            .setBackups(2)
    }
}
